tutorials/tour-of-restate-rust/src/cart_object.rs

```diff
@@ -0,0 +1,98 @@
+use crate::checkout_service::{CheckoutRequest, CheckoutServiceClient};
+use crate::ticket_object::TicketObjectClient;
+use restate_sdk::prelude::*;
+use std::time::Duration;
+
+#[restate_sdk::object]
+pub(crate) trait CartObject {
+    #[name = "addTicket"]
+    async fn add_ticket(ticket_id: String) -> Result<bool, HandlerError>;
+    async fn checkout() -> Result<bool, HandlerError>;
+    #[name = "expireTicket"]
+    async fn expire_ticket(ticket_id: String) -> Result<(), HandlerError>;
+}
+
+pub struct CartObjectImpl;
+
+const TICKETS: &str = "ticket";
+
+impl CartObject for CartObjectImpl {
+    async fn add_ticket(
+        &self,
+        ctx: ObjectContext<'_>,
+        ticket_id: String,
+    ) -> Result<bool, HandlerError> {
+        let reservation_success = ctx
+            .object_client::<TicketObjectClient>(ticket_id.clone())
+            .reserve()
+            .call()
+            .await?;
+
+        if reservation_success {
+            let mut tickets = ctx
+                .get::<Json<Vec<String>>>(TICKETS)
+                .await?
+                .unwrap_or_default()
+                .into_inner();
```

Comment:
this looks so bad. I wonder if i can make this api a tad bit better :(

Reply:
yeah it's quite verbose